File: modules/076-skills-management-system/tools2/scripts/src/commands/root.ts
import { runInit } from './init';
import { runPull } from './pull';
import { runPush } from './push';
import { runList } from './list';
import { runCreate } from './create';
import { runEnable, runDisable } from './toggle';
import { runAIHelp } from './aihelp';
import { runInitRepo } from './initrepo';
import { printHelp } from './help';

export function execute(args: string[]): void {
  if (args.length === 0 || args[0] === 'help' || args[0] === '--help' || args[0] === '-h') {
    printHelp();
    return;
  }

  switch (args[0]) {
    case 'init':
      runInit(args.slice(1));
      break;
    case 'pull':
      runPull(args.slice(1));
      break;
    case 'push':
      runPush(args.slice(1));
      break;
    case 'list':
      runList(args.slice(1));
      break;
    case 'create':
      runCreate(args.slice(1));
      break;
    case 'enable':
      runEnable(args.slice(1));
      break;
    case 'disable':
      runDisable(args.slice(1));
      break;
    case 'ai-help':
      runAIHelp();
      break;
    case 'init-repo':
      runInitRepo(args.slice(1));
      break;
    default:
      console.error(`Error: unknown command "${args[0]}"\n`);
      printHelp();
      process.exit(1);
  }
}

File: modules/076-skills-management-system/tools_py/skills_cli/commands/root.py
"""Command dispatch for the skills CLI."""

import sys
from typing import Sequence

from ..lib.errors import CommandError
from .help import print_help


def execute(args: Sequence[str]) -> int:
    """Dispatch command-line arguments and return a process exit code."""
    if not args or args[0] in {"help", "--help", "-h"}:
        print_help()
        return 0

    command = args[0]
    try:
        if command == "init":
            from .init import run_init

            run_init(list(args[1:]))
        elif command == "pull":
            from .pull import run_pull

            run_pull(list(args[1:]))
        elif command == "push":
            from .push import run_push

            run_push(list(args[1:]))
        elif command == "list":
            from .list import run_list

            run_list(list(args[1:]))
        elif command == "create":
            from .create import run_create

            run_create(list(args[1:]))
        elif command == "enable":
            from .toggle import run_enable

            run_enable(list(args[1:]))
        elif command == "disable":
            from .toggle import run_disable

            run_disable(list(args[1:]))
        elif command == "ai-help":
            from .aihelp import run_ai_help

            run_ai_help()
        elif command == "init-repo":
            from .initrepo import run_init_repo

            run_init_repo(list(args[1:]))
        else:
            print(f'Error: unknown command "{command}"', file=sys.stderr)
            print(file=sys.stderr)
            print_help()
            return 1
    except CommandError as exc:
        message = str(exc)
        if not message.startswith("Error:"):
            message = f"Error: {message}"
        print(message, file=sys.stderr)
        return 1
    except Exception as exc:
        print(f"Error: {exc}", file=sys.stderr)
        return 1
    return 0
